test/scimPatch.test.ts

```diff
@@ -544,6 +544,15 @@ describe('SCIM PATCH', () => {
             return done();
         });
 
+        it('REMOVE: nested array element with value supplied', done => {
+            scimUser.name.nestedArray = [{primary: true, value: 'value1'}, {primary: false, value: 'value2'}];
+            const patch1: ScimPatchRemoveOperation = {op: 'remove', path: 'name.nestedArray', value: [{value: 'value2', primary: false}]};
```

Comment:
I am not understanding the RFC like you for this part.
> If the target location is a multi-valued attribute and a complex filter is specified comparing a "value", the values matched by the filter are removed.  If no other values remain after removal of the selected values, the multi-valued attribute SHALL be considered unassigned.

For me, we should not have an array for the value but directly the object.
Something like that 
```typescript
const patch1: ScimPatchRemoveOperation = {op: 'remove', path: 'name.nestedArray', value: {value: 'value2', primary: false}};
```
And in that case, your code is falling because the function `filterWithArray<T>(arr: T[], itemsToRemove: T[]): T[]` supposed that the value is always an Array.

Reply:
It's a live example from OneLogin, it sends array of objects to remove. I want to go with it for now, and maybe add a straight extension for 1 object entry